src/nvapi_sys.cpp

```diff
@@ -89,4 +89,48 @@ extern "C" {
 
         return Ok(n);
     }
+
+    NvAPI_Status __cdecl NvAPI_SYS_GetPhysicalGPUs(NV_PHYSICAL_GPUS* pPhysicalGPUs) {
+        constexpr auto n = __func__;
+
+        if (log::tracing())
+            log::trace(n, log::fmt::ptr(pPhysicalGPUs));
+
+        if (nvapiAdapterRegistry == nullptr)
+            return ApiNotInitialized(n);
+
+        if (pPhysicalGPUs == nullptr)
+            return InvalidArgument(n);
+
+        for (auto i = 0U; i < nvapiAdapterRegistry->GetAdapterCount(); i++) {
+            pPhysicalGPUs->gpuHandleData[i].hPhysicalGpu = reinterpret_cast<NvPhysicalGpuHandle>(nvapiAdapterRegistry->GetAdapter(i));
+            pPhysicalGPUs->gpuHandleData[i].adapterType = NV_ADAPTER_TYPE_UNKNOWN;
```

Comment:
I think it would be safer to report `NV_ADAPTER_TYPE_WDDM` instead, afaik it's the model you expect to see when you want to render some stuff and present it to the screen, so it should be the best match for GPUs that aren't in data centers or some compute-only clusters. Or call NVML to check but be aware that `wine-nvml` will just always return WDDM :stuck_out_tongue:

Same in `NvAPI_SYS_GetLogicalGPUs`.

Reply:
Makes sense.